components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -93,7 +111,79 @@ function setSelectedId() {
+    const facilitySections = facilitiesStore.facilitySectionFields as unknown as {
+        [key: string]: string
+    }
+
+    const areThereUnsavedChanges
+        = facilityBeforeChange.nameEn !== facilitySections.nameEn
+        || facilityBeforeChange.nameJa !== facilitySections.nameJa
+        || facilityBeforeChange.contact.phone !== facilitySections.phone
+        || facilityBeforeChange.contact.website !== facilitySections.website
+        || facilityBeforeChange.contact.email !== facilitySections.email
+        || facilityBeforeChange.contact.address.postalCode !== facilitySections.postalCode
+        || facilityBeforeChange.contact.address.prefectureEn !== facilitySections.prefectureEn
+        || facilityBeforeChange.contact.address.cityEn !== facilitySections.cityEn
+        || facilityBeforeChange.contact.address.addressLine1En !== facilitySections.addressLine1En
+        || facilityBeforeChange.contact.address.addressLine2En !== facilitySections.addressLine2En
+        || facilityBeforeChange.contact.address.prefectureJa !== facilitySections.prefectureJa
+        || facilityBeforeChange.contact.address.cityJa !== facilitySections.cityJa
+        || facilityBeforeChange.contact.address.addressLine1Ja !== facilitySections.addressLine1Ja
+        || facilityBeforeChange.contact.address.addressLine2Ja !== facilitySections.addressLine2Ja
+        || facilityBeforeChange.contact.googleMapsUrl !== facilitySections.googlemapsURL
+        || facilityBeforeChange.mapLatitude.toString() !== facilitySections.mapLatitude
+        || facilityBeforeChange.mapLongitude.toString() !== facilitySections.mapLongitude
+        || JSON.stringify(facilityBeforeChange.healthcareProfessionalIds)
+        !== JSON.stringify(facilitySections.healthcareProfessionalIds)
+        || facilitySections.healthProfessionalsRelations.length > 0
+
+    return areThereUnsavedChanges
+}
+
+const updateFacilityOrHealthcareProfessional = async () => {
+    // This makes the on click update the facility if the screen is EditFacility
+    if (moderationScreenStore.activeScreen === ModerationScreen.EditFacility) {
+        const checkForUnsavedChanges = facilityHasUnsavedChanges()
+
+        // This prevents us from sending a requested unnecessarily if the user has not made changes
+        if (!checkForUnsavedChanges) return
+
+        const response = await facilitiesStore.updateFacility()
+
+        if (response.errors?.length) {
+            handleServerErrorMessaging(response.errors, toast, t)
+            return response
+        }
+
+        facilitiesStore.initializeFacilitySectionValues(response.data as Facility)
+        toast.success(t('modEditFacilityOrHPTopbar.facilityUpdatedSuccessfully'))
+        return response
+    }
+}
+
+const updateFacilityOrHealthcareProfessionalAndExit = async () => {
+    const response = await updateFacilityOrHealthcareProfessional()
+
+    if (response && response.errors?.length) {
+        return
```

Comment:
silent exception! let's at least log to console. eventually this should be a log

Reply:
there is always an error logged to the console with the api call. I added the toast messaging that should have been added